### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by belmeetmule**
- [ ] set up linters
- [ ] set gitflow
- [ ] add person, student and teacher classes and files
- [ ] Used inheritance to create drived classes from parent class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @name = name
+    @parent_permission = parent_permission
+  end
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def of_age?
+    @age >= 18
+  end
+
```
Comment (HAMMAS-SALEEM) [inline person.rb:14]: - [ ] This method is private which is a requirement so please make it private :blush:

Hint: You can make it private by adding the following line at line 14 :+1:
**private of_age?**

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  def can_use_services?
+    if of_age? || @parent_permission
+      true
+    else
+      false
+    end
+  end
```
Comment (HAMMAS-SALEEM) [inline person.rb]: - [ ] You have made this function but you could have made it in less code which is shown below :+1:

```
def can_use_services?
    of_age? || @parent_permission
end
```

File: README.md [hunk 3]
```diff
@@ -1 +1,27 @@
...
+
+#### 3 - run the project
+> Try it out in your IRB console by instantiating any of the classes
+
+## Author
+### 👨‍💻 Mulugeta M.
+
+- GitHub: [@belmeetmule](https://github.com/belmeetmule)
+- Twitter: [@belmeetmule](https://twitter.com/belmeetmule)
+- LinkedIn: [belmeetmule](https://linkedin.com/in/belmeetmule)
```
Comment (HAMMAS-SALEEM) [inline README.md]: image

- [ ] You didn't add these sections in your readme file shown in the image above :arrow_up: Kindly add the missing section using this [microverse readme template](https://github.com/microverseinc/readme-template#-contributing) :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @belmeetmule :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @belmeetmule,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**